shards/gfx/context_data.hpp

```diff
@@ -42,7 +42,11 @@ struct ContextData : public std::enable_shared_from_this<ContextData> {
 template <typename T> struct TWithContextData {
   std::shared_ptr<T> contextData;
 
-  T &createContextDataConditional(Context &context) {
+  T &createContextDataConditionalRefUNSAFE(Context &context) {
```

Comment:
Only safe if texture is not updated from another thread ever (internal stuff is fine, shards touched stuff not always)